include/deal.II/matrix_free/matrix_free.h

```diff
@@ -243,7 +243,9 @@ class MatrixFree : public Subscriptor
       , hold_all_faces_to_owned_cells(hold_all_faces_to_owned_cells)
       , cell_vectorization_categories_strict(
           cell_vectorization_categories_strict)
-    {}
+    {
+      this->communicator_sm = MPI_COMM_SELF;
```

Comment:
Why don't you include `communicator_sm(MPI_COMM_SELF)` in the list above for symmetry with the other parameters?